plasma_framework/truffle-config.js

```diff
@@ -24,7 +24,7 @@ module.exports = {
             provider: () => new HDWalletProvider(
                 [
                     process.env.DEPLOYER_PRIVATEKEY || '0'.repeat(64),
-                    process.env.MAINTAINER_PRIVATEKEY || '0'.repeat(64),
+                    fs.readFileSync('../MultiSigWallet/build/multisig_instance') || process.env.MAINTAINER_PRIVATEKEY || '0'.repeat(64),
```

Comment:
I think this is misleading....you get deployed contract address from the file right (not private key...and there is no way to get a private key of the contract)?

This is one reason that is why I was suggesting a restructure of this comment: https://github.com/omgnetwork/plasma-contracts/pull/649#pullrequestreview-461371869

Reply:
yeah... I don't think this is used at all :D

Reply:
lemme review what you said in #649

Reply:
okay... https://github.com/omgnetwork/plasma-contracts/pull/649#pullrequestreview-461371869
to regards to your comment. The comment I think isn't really related to the issue at hand and I'm not sure how it's a better approach regardless. It seems (the current way) is a fairly standard way how people provide accounts to truffle migration. Having them in config doesn't really change anything much. Or am I not understanding something?

Reply:
> and I'm not sure how it's a better approach regardless.

you do not need such `fs.readFileSync('../MultiSigWallet/build/multisig_instance')` thing (well...you do not need this line here too tbh) also all the [branching of getting vault address](https://github.com/omgnetwork/plasma-contracts/blob/bbdb558b4f61c54e34a7d8342c412ed45c13f3ec/plasma_framework/migrations/20_deploy_plasma_framework.js#L11-L17) could be hidden to a single place, reverting the dependency control of vault address to higher level of truffle migration script. In this case we have 2 options, `truffle-config.js` or `config.js`. I am not sure how to do that via `truffle-config.js` so I proposed `config.js`

> is a fairly standard way 

true, but current standard tooling (hd-wallet-provider) does not support what you need. That is what causing the complexity here.

hmmm...however, you have point me to a direction that it might be possible to just implement a new provider that would link with different way to "sending/approving transaction". Did a bit research and have not find enough documentation on this yet. So just throwing out idea at this moment. I will side track this direction differently and can/should be out of your effort of delivering this (unless it becomes obvious and it is super easy to do)